src/python/pants/backend/helm/util_rules/sources.py

```diff
@@ -76,18 +107,55 @@ def valid_sources_types(self) -> tuple[type[SourcesField], ...]:
             types.append(FileSourceField)
         return tuple(types)
 
+    def debug_hint(self) -> str | None:
+        return self.field_set.address.spec
+
 
 @dataclass(frozen=True)
 class HelmChartSourceFiles:
     snapshot: Snapshot
+    unrooted_files: tuple[str, ...]
+
+
+@rule_helper
+async def _strip_chart_source_root(source_files: SourceFiles, source_root: SourceRoot) -> Snapshot:
```

Comment:
Do you need this at all now?

Reply:
yeah, there are two places in which I need to remove 'the prefix'  on the sources, one of them is for first party charts (`helm_chart` targets) which is done now by auto-detecting the source root.

The other one is for `helm_artifact` targets, which are basically tarballs that we download and uncompress.